usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -99,6 +99,44 @@ def post(self, request):
         return Response(response)
 
 
+class CFDAAutocompleteViewSet(BaseAutocompleteViewSet):
+
+    def post(self, request):
+        """Return all budget function/subfunction titles matching the provided search text"""
+
+        search_text, limit = self.get_request_payload(request)
+
+        # get relevant TransactionContracts
+        queryset = Cfda.objects.filter(program_number__isnull=False,
+                                       program_title__isnull=False)
+        # Filter based on search text
+        response = {}
+
+        queryset = queryset.annotate(similarity=Greatest(
+            TrigramSimilarity('program_number', search_text),
+            TrigramSimilarity('program_title', search_text),
+            TrigramSimilarity('popular_name', search_text)))\
+            .distinct().order_by('-similarity')
+
+        queryset = queryset.annotate(prog_num_similarity=TrigramSimilarity('program_number', search_text))
+
```

Comment:
there is a posseble error error with trigram search where "2323" will return a 100% match with "232323" so if there is an 100% match on a number, i added a length check.